lib/trino-parquet/src/test/java/io/trino/parquet/writer/TestParquetWriter.java

```diff
@@ -348,4 +364,82 @@ public void testDictionaryPageOffset()
             }
         }
     }
+
+    @Test
+    public void testWriteBloomFilters()
+            throws IOException
+    {
+        List<String> columnNames = ImmutableList.of("columnInteger", "columnBigint", "columnReal", "columnDouble", "columnVarbinary", "columnVarchar", "columnUUID");
+        List<Type> types = ImmutableList.of(INTEGER, BIGINT, REAL, DOUBLE, VARBINARY, VARCHAR,
+                // This should be UUID, but we can only read, not write UUID natively
+                VARBINARY);
+
+        int size = 2000;
+        Random random = new Random(42);
+        List<List<?>> data = ImmutableList.of(
+                shuffle(random, LongStream.range(0, size).boxed().toList()),
+                shuffle(random, LongStream.range(0, size).boxed().toList()),
+                shuffle(random, LongStream.range(0, size).boxed().map(x -> (long) floatToRawIntBits((float) x)).toList()),
+                shuffle(random, LongStream.range(0, size).asDoubleStream().boxed().toList()),
+                shuffle(random, LongStream.range(0, size).boxed().map(i -> Slices.utf8Slice(i.toString())).toList()),
+                shuffle(random, LongStream.range(0, size).boxed().map(i -> Slices.utf8Slice(i.toString())).toList()),
+                shuffle(random, LongStream.range(0, size).boxed().map(i -> javaUuidToTrinoUuid(new java.util.UUID(i, i))).toList()));
+
+        ParquetDataSource dataSource = new TestingParquetDataSource(
+                writeParquetFile(
+                        ParquetWriterOptions.builder()
+                                .setMaxBlockSize(DataSize.ofBytes(64 * 1024))
+                                .setBloomFilterColumns(columnNames.stream().collect(toImmutableSet()))
+                                .build(),
+                        types,
+                        columnNames,
+                        generateInputPages(types, 100, data)),
+                new ParquetReaderOptions());
+
+        ParquetMetadata parquetMetadata = MetadataReader.readFooter(dataSource, Optional.empty());
+        // Check that bloom filters are right after each other at the end of each row group
+        for (BlockMetaData block : parquetMetadata.getBlocks()) {
+            assertThat(block.getColumns().get(1).getBloomFilterOffset() - block.getColumns().get(0).getBloomFilterOffset())
+                    .isEqualTo(1024 * 1024 + 18);
```

Comment:
How is this number derived ?
It seems to me we should check that "offset + size" is equal to next "offset"

Reply:
This is bitset size + metadata size. The metadata is written with Thrift, so its size depends on the size of the bitset.